x-pack/plugins/aiops/public/embeddables/change_point_chart/embeddable_change_point_chart_factory.tsx

```diff
@@ -0,0 +1,229 @@
+      };
+
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        changePointControlsApi,
+        changePointControlsComparators,
+        serializeChangePointChartState,
+      } = initializeChangePointControls(state);
+
+      const dataLoading = new BehaviorSubject<boolean | undefined>(true);
+      const blockingError = new BehaviorSubject<Error | undefined>(undefined);
+
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>([
+        await deps.data.dataViews.get(state.dataViewId),
+      ]);
+
+      const api = buildApi(
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...changePointControlsApi,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.aiops.changePointDetection.typeDisplayName', {
+              defaultMessage: 'change point charts',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableChangePointUserInput } = await import(
+                './handle_explicit_input'
+              );
+
+              const result = await resolveEmbeddableChangePointUserInput(
+                coreStart,
+                pluginStart,
+                serializeChangePointChartState()
+              );
+
+              changePointControlsApi.updateUserInput(result);
+            } catch (e) {
+              return Promise.reject();
+            }
+          },
+          dataLoading,
+          blockingError,
+          dataViews: dataViews$,
+          serializeState: () => {
+            return {
+              rawState: {
+                timeRange: undefined,
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeChangePointChartState(),
+              },
+              references: [],
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...changePointControlsComparators,
+        }
+      );
+
+      const ChangePointDetectionComponent = getChangePointDetectionComponent(
+        coreStart,
+        pluginStart
+      );
+
+      return {
+        api,
+        Component: () => {
+          if (!apiHasExecutionContext(parentApi)) {
+            throw new Error('Parent API does not have execution context');
+          }
+
+          const [dataViewId, viewType, fn, metricField, splitField, maxSeriesToPlot, partitions] =
+            useBatchedPublishingSubjects(
+              api.dataViewId,
+              api.viewType,
+              api.fn,
+              api.metricField,
+              api.splitField,
+              api.maxSeriesToPlot,
+              api.partitions
+            );
+
+          const reload$ = useMemo(
```

Comment:
This is a neat way to translate the fetch pipe into a `lastReloadRequestTime`. If this ends up being a common pattern, we could make a helper for it - something like `useFetch(api)` - which would return _all_ of the fetch context as reactive vars.